redis_wrapper.go

```diff
@@ -11,89 +10,62 @@ type RedisWrapper struct {
 }
 
-func (wrapper RedisWrapper) Set(key string, value string, expiration time.Duration) bool {
-	return checkErr(wrapper.rawClient.Set(key, value, expiration).Err())
+func (wrapper RedisWrapper) Set(key string, value string, expiration time.Duration) error {
+	// NOTE: using Err() here because Result() string is always "OK"
+	return wrapper.rawClient.Set(key, value, expiration).Err()
 }
 
-func (wrapper RedisWrapper) Del(key string) (affected int, ok bool) {
-	n, err := wrapper.rawClient.Del(key).Result()
-	ok = checkErr(err)
-	if !ok {
-		return 0, false
-	}
-	return int(n), ok
+func (wrapper RedisWrapper) Del(key string) (affected int64, err error) {
+	return wrapper.rawClient.Del(key).Result()
 }
 
-func (wrapper RedisWrapper) TTL(key string) (ttl time.Duration, ok bool) {
-	ttl, err := wrapper.rawClient.TTL(key).Result()
-	ok = checkErr(err)
-	if !ok {
-		return 0, false
-	}
-	return ttl, ok
+func (wrapper RedisWrapper) TTL(key string) (ttl time.Duration, err error) {
+	return wrapper.rawClient.TTL(key).Result()
 }
 
-func (wrapper RedisWrapper) LPush(key string, value ...string) bool {
-	return checkErr(wrapper.rawClient.LPush(key, value).Err())
-}
-
-func (wrapper RedisWrapper) LLen(key string) (affected int, ok bool) {
-	n, err := wrapper.rawClient.LLen(key).Result()
-	ok = checkErr(err)
-	if !ok {
-		return 0, false
-	}
-	return int(n), ok
+func (wrapper RedisWrapper) LPush(key string, value ...string) (total int64, err error) {
+	return wrapper.rawClient.LPush(key, value).Result()
 }
 
-func (wrapper RedisWrapper) LRem(key string, count int, value string) (affected int, ok bool) {
-	n, err := wrapper.rawClient.LRem(key, int64(count), value).Result()
-	return int(n), checkErr(err)
+func (wrapper RedisWrapper) LLen(key string) (affected int64, err error) {
+	return wrapper.rawClient.LLen(key).Result()
 }
 
-func (wrapper RedisWrapper) LTrim(key string, start, stop int) {
-	checkErr(wrapper.rawClient.LTrim(key, int64(start), int64(stop)).Err())
+func (wrapper RedisWrapper) LRem(key string, count int64, value string) (affected int64, err error) {
+	return wrapper.rawClient.LRem(key, int64(count), value).Result()
 }
 
-func (wrapper RedisWrapper) RPopLPush(source, destination string) (value string, ok bool) {
-	value, err := wrapper.rawClient.RPopLPush(source, destination).Result()
-	return value, checkErr(err)
+func (wrapper RedisWrapper) LTrim(key string, start, stop int64) error {
+	// NOTE: using Err() here because Result() string is always "OK"
+	return wrapper.rawClient.LTrim(key, int64(start), int64(stop)).Err()
 }
 
-func (wrapper RedisWrapper) SAdd(key, value string) bool {
-	return checkErr(wrapper.rawClient.SAdd(key, value).Err())
+func (wrapper RedisWrapper) RPopLPush(source, destination string) (value string, err error) {
+	value, err = wrapper.rawClient.RPopLPush(source, destination).Result()
+	// println("RPopLPush", source, destination, value, err)
+	switch err {
+	case nil:
+		return value, nil
+	case redis.Nil:
```

Comment:
why we only check `redis.Nil` for this function? shall we also check it for the others?

Reply:
I checked the documentation and tried all the Redis commands we use in `redis-cli` and surprisingly `RPopLPush` is the only one that can actually return a `redis.Nil` error.